datafusion/optimizer/src/push_down_filter.rs

```diff
@@ -881,6 +910,42 @@ pub fn replace_cols_by_name(
     })
 }
 
+/// check whether the expression is volatile predicates
+pub fn is_volatile_expression(e: Expr) -> bool {
```

Comment:
```suggestion
fn is_volatile_expression(e: &Expr) -> bool {
```
- We can pass the argument `e` by reference.
- I would suggest setting the visibility of this function to `private` since it is only being used within this file.